ts/sagas/wallet.ts

```diff
@@ -497,46 +503,49 @@ function* completionHandler(_: PaymentRequestCompletion) {
   yield put({ type: PAYMENT_COMPLETED });
 }
 
-function* fetchPagoPaToken(pagoPaClient: PagoPaClient): Iterator<Effect> {
-  const token: Option<string> = yield select(walletTokenSelector);
-  if (token.isSome()) {
-    const response: SagaCallReturnType<
-      typeof pagoPaClient.getSession
-    > = yield call(pagoPaClient.getSession, token.value);
-    if (response !== undefined && response.status === 200) {
-      // token fetched successfully, store it
-      yield put(storePagoPaToken(some(response.value.data.sessionToken)));
-    }
+function* fetchPagoPaToken(
+  pagoPaClient: PagoPaClient,
+  walletToken: string
+): Iterator<Effect> {
+  const response: SagaCallReturnType<
+    typeof pagoPaClient.getSession
+  > = yield call(pagoPaClient.getSession, walletToken);
+  if (response !== undefined && response.status === 200) {
+    // token fetched successfully, store it
+    yield put(storePagoPaToken(some(response.value.data.sessionToken)));
   }
```

Comment:
btw there doesn't seem to be any logic that handles a failure here?

Reply:
If the token cannot be loaded successfully here, the retry logic (that is implemented by the caller) will reach the maximum number of attempts and then fail (the failure logic at the caller-level still needs to be implemented as a part of a different story). Handling the failure here would not result in much (the error would need to be propagated to the caller anyway)